deployment/SampleClaimsProvider/ClaimsProviderStore.cs

```diff
@@ -0,0 +1,468 @@
+        {
+            Requires.Argument(NamingRestrictions.IsValidNamingString(vehicleId), nameof(vehicleId), NamingRestrictions.InvalidNamingStringErrorMessage);
+
+            if (userId != null)
+            {
+                Requires.Argument(NamingRestrictions.IsValidNamingString(userId), nameof(userId), NamingRestrictions.InvalidNamingStringErrorMessage);
+            }
+
+            IDictionary<string, List<StringClaim>> labeledClaims = new Dictionary<string, List<StringClaim>>();
+            bool isVehicleUserClaimsPresent = false;
+
+            // If an authToken is provided in the request, verify and determine how the authToken will be used (for example JIT elevation) and what claims to provide.
+            if (authToken?.Length > 0)
+            {
+                IDictionary<string, IList<StringClaim>> labeledAuthClaims = this.LookupClaimsByAuthToken(authToken, labelsAndPaths);
+                this.AppendToExistingLabeledClaims(labeledClaims, labeledAuthClaims);
+            }
+
+            if (userId?.Length > 0)
+            {
+                // Aggregate the claims from Claims database for the VehicleId/UserId, VehicleId, UserId
+                IDictionary<string, IList<StringClaim>> userVehicleClaims = await this.LookupClaimsByVehicleAndUser(vehicleId, userId, labelsAndPaths);
+                if (userVehicleClaims?.Count > 0)
+                {
+                    isVehicleUserClaimsPresent = true;
+
+                    this.AppendToExistingLabeledClaims(labeledClaims, userVehicleClaims);
+                }
+
+                // If there are no claims at this point - there is no association of userId with the vehicleId
+                if (labeledClaims.Count == 0)
+                {
+                    return null;
+                }
+
+                // Claims associated with the user
+                IDictionary<string, IList<StringClaim>> userClaims = await this.LookupClaimsByVehicleAndUser(null, userId, labelsAndPaths);
+                this.AppendToExistingLabeledClaims(labeledClaims, userClaims);
+            }
+
+            // Claims associated with the vehicle
+            IDictionary<string, IList<StringClaim>> vehicleClaims = await this.LookupClaimsByVehicleAndUser(vehicleId, null, labelsAndPaths);
+            if (vehicleClaims?.Count > 0)
+            {
+                this.AppendToExistingLabeledClaims(labeledClaims, vehicleClaims);
+            }
+            else
+            {
+                // if there were no vehicle specific claims (either from Vehicle/User claims or Vehicle only claims), this means the Vehicle is not in the database so fail the request
+                if (vehicleClaims == null && !isVehicleUserClaimsPresent)
+                {
+                    return null;
+                }
+            }
```

Comment:
All of `LookupVehicleUserAsync` can be simplified greatly, especially if it no longer deals with labeled claims and only works with `IList<StringClaim>`.

See my implementation in the removed version of this function for an idea of how. The removed implementation is missing two small parts, the authtoken claims and getting claims associated with just the userid and `null` vehicleid. 

We can remove the entire `AppendToExistingLabeledClaims` function and just use `AddRange` to merge claims.
The `isVehicleUserClaimsPresent` boolean is unnecessary, we can know if we have vehicleUserClaims if our `userVehicleClaims` list isn't empty.